src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,29 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithForm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._form = this._popupElement.querySelector(".modal__form");
+    this._handleFormSubmit = handleFormSubmit;
+    this._modalButton = this._popupElement.querySelector(".modal__button");
+  }
+  setSubmit(handleSubmit) {
+    this._handleFormSubmit = handleSubmit;
+  }
+
+  setLoading(isLoading) {
+    if (isLoading) {
+      this._modalButton.textContent = "Saving...";
+    } else {
+      this._modalButton.textContent = "Save";
```

Comment:
You need to return the correct button text for the confrimation modal. It's `Yes`